### PR TITLE
Add UpDownBase tests and related Control tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Add UpDownBase tests and related Control tests

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2457)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs [hunk 1]
```diff
@@ -103,6 +117,12 @@ public void AxHost_Ctor_String(string clsid)
         public void AxHost_Ctor_String_Int(string clsid, int flags)
         {
             var control = new SubAxHost(clsid, flags);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AxHostTests.cs]: ditto

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs [hunk 2]
```diff
@@ -41,13 +47,17 @@ public void ComponentEditorPagePanel_Ctor_Default()
...
+            Assert.False(control.CanFocus);
             Assert.True(control.CanRaiseEvents);
+            Assert.False(control.CanSelect);
+            Assert.False(control.Capture);
             Assert.True(control.CausesValidation);
             Assert.Equal(new Rectangle(0, 0, 200, 100), control.ClientRectangle);
             Assert.Equal(new Size(200, 100), control.ClientSize);
             Assert.False(control.CommitOnDeactivate);
-            Assert.Null(control.Component);
+            Assert.Null(control.Component);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs]: 🙈

File: src/System.Windows.Forms/src/System/Windows/Forms/UpDownBase.cs [hunk 3]
```diff
@@ -851,7 +851,6 @@ protected virtual void OnTextBoxTextChanged(object source, EventArgs e)
         {
             if (changingText)
             {
-                Debug.Assert(UserEdit == false, "OnTextBoxTextChanged() - UserEdit == true");
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/UpDownBase.cs]: Why is this being removed?
There are two flags `UserEdit` and `ChangingText` that are mutually exclusive, and this assert helps detecting if they are being set simultaneously.

Whilst `DomainUpDown` guards against it
https://github.com/dotnet/winforms/blob/5aeb34a3075e321ec3dc96c067e28ecdda3d33ee/src/System.Windows.Forms/src/System/Windows/Forms/DomainUpDown.cs#L576-L579
`NumericUpDown` doesn't:
https://github.com/dotnet/winforms/blob/5aeb34a3075e321ec3dc96c067e28ecdda3d33ee/src/System.Windows.Forms/src/System/Windows/Forms/NumericUpDown.cs#L785-L798

A better way perhaps would be to express this as a single flag with an enumerated value, but it is what it is now.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 4]
```diff
@@ -501,7 +501,6 @@ public AccessibleObject AccessibilityObject
                     // ControlAccessibleObject and not some other derived class
                     if (!(accessibleObject is ControlAccessibleObject))
                     {
-                        Debug.Fail("Accessible objects for controls must be derived from ControlAccessibleObject.");
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: Why is this being removed?

/cc: @M-Lipin @vladimir-krestov

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: Can happen in following test
```cs

public static IEnumerable<object[]> AccessibilityObject_CustomCreateAccessibilityInstance_TestData()
{
    yield return new object[] { null, null };
    yield return new object[] { new AccessibleObject(), null };

    var controlAccessibleObject = new Control.ControlAccessibleObject(new Control());
    yield return new object[] { controlAccessibleObject, controlAccessibleObject };
}

[WinFormsTheory]
[MemberData(nameof(AccessibilityObject_CustomCreateAccessibilityInstance_TestData))]
public void Control_AccessibilityObject_GetCustomCreateAccessibilityInstance_ReturnsExpected(AccessibleObject result, AccessibleObject expected)
{
    using var control = new CustomCreateAccessibilityInstanceControl
    {
        CreateAccessibilityResult = result
    };
    Assert.Same(expected, control.AccessibilityObject);
    Assert.Same(control.AccessibilityObject, control.AccessibilityObject);
    Assert.False(control.IsHandleCreated);
}

private class CustomCreateAccessibilityInstanceControl : Control
{
    public AccessibleObject CreateAccessibilityResult { get; set; }

    protected override AccessibleObject CreateAccessibilityInstance() => CreateAccessibilityResult;
}
```

File: src/System.Windows.Forms/src/System/Windows/Forms/UpDownBase.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/UpDownBase.cs]: OK I added NoAssertContext.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 4 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: Sorry @hughbe, but this is a very bad example, and this is precisely the situation this assert is guarding against.
We must not modify our production code to fit unit tests.

In these tests we expect to go through the `return null` code path, and thus expect to fail on the assertion - we need to employ `NoAssertContext` to mitigate the crash.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: Sure will fix

**Comment by RussKie** [comment]
Thank you

**Pull request merged**